code/datums/datumvars.dm

```diff
@@ -1,441 +1,517 @@
 // reference: /client/proc/modify_variables(var/atom/O, var/param_var_name = null, var/autodetect_class = 0)
 
 /datum
-	var/var_edited = 0 //Warrenty void if seal is broken
-
-/datum/proc/on_varedit(modified_var) //called whenever a var is edited
-	var_edited = 1
+	var/var_edited = FALSE //Warranty void if seal is broken
+
+/datum/proc/can_vv_get(var_name)
+	return TRUE
+
+/client/proc/can_vv_get(var_name)
+	return TRUE
+
+/datum/proc/vv_edit_var(var_name, var_value) //called whenever a var is edited

```

Comment:
If we used this [here](https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/admin/verbs/SDQL2/SDQL_2.dm#L133), we could ditch the `datum_is_forbidden` system entirely and define restricted vars directly on the objects themselves